### ISSUE TITLE
ReflectionClass throws error with Map collection as default parameter

### ISSUE THREAD

**Issue by kyleferguson** [facebook/hhvm#5405]
If the signature of a method has a Map collection as the default value of a parameter, an error is thrown when using the ReflectionClass to inspect that method.

```
Cannot use collection initialization in non-collection class
```

For example, with a class:

```
<?hh //strict
namespace Acme;

class Foo
{
    protected Map<string, string> $options;

    public function __construct(Map<string, string> $options = Map{})
    {
        $this->options = $options;
    }

}
```

then using reflection like so will throw the error:

```
$reflection = new ReflectionClass('Acme\Foo');

$reflection->getConstructor()->getParameters();
```

Instantiating the class and using it like normal works just fine. The only issue I've seen is when trying to inspect the method using reflection. All other values/types work well with the ReflectionClass in my experience.

**Comment by jwatzman** [comment]
Yep, I can repro with this all-in-one test file:

```hh
<?hh

namespace Acme {
  class Foo {
    protected Map<string, string> $options;

    public function __construct(Map<string, string> $options = Map {}) {
      $this->options = $options;
    }
  }
}

namespace {
  $reflection = new ReflectionClass('Acme\Foo');
  $reflection->getConstructor()->getParameters();
}
```

**Comment by jwatzman** [comment]
Someone will take a look when they get a chance, thanks for reporting!

**Comment by kyleferguson** [comment]
In case it helps, I also found that if you do all of that without namespacing, the issue goes away.

**Comment by lexidor** [comment]
This issue was fixed in [4.66.0](https://hhvm.com/blog/2020/07/13/hhvm-4.66.html).

If a function has a parameter with an invalid default value, accessing it via ReflectionFunction or ReflectionMethod may currently throw. In the future, it will only raise a warning (making reflection safer to use).
 - Use the INI option hhvm.fix_default_arg_reflection=2 to enable the future behavior now.

_Failed to eval default value ... assigning NULL instead_

It appears to be a conscious decision made by the people working on the runtime to disallow this.
